simulationcraft/simc
# [Demon Hunter] Refactor metamorphosis_buff_t

For review/testing by @Zarania 

[Demon Hunter] Refactor metamorphosis_buff_t
* Merge Havoc and Vengeance to use the same buff creation process
* Add tracking for Demonic triggers to prefer re-triggers by consecutive non-Metamorphosis Eye Beams
* Add buff.metamorphosis.triggered_by_demonic expression to build APL logic based on above functionality
* Call adjust_recharge_multiplier on cooldown.blade_dance when triggering Metamorphosis to ensure Death Sweep benefits from Metamorphosis haste if already on cooldown
* Add 7.1.5 armor buff for Vengeance to Metamorphosis
